src/inference/src/dev/itensor.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+
+#include "openvino/core/coordinate.hpp"
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/runtime/allocator.hpp"
+
+namespace InferenceEngine {
+
+class Blob;
+
+}
+
+namespace ov {
+
+class ITensor : public std::enable_shared_from_this<ITensor> {
+public:
+    /**
+     * @brief Set new shape for tensor, deallocate/allocate if new total size is bigger than previous one.
```

Comment:
Will it be less strict than current `ov::Tensor`? 
Where set new shape was not possible if layout was not same e.g: 
tensor shape is `{0, 0}` then set to new `{10, 20}` is possible but set to `{10,20,2}` triggered exception.